site/docs/hosting/vps.md

```diff
@@ -2,24 +2,129 @@
 
 A virtual private server, mostly known as VPS, is a virtual machine running in the cloud with its users having the full control of its system.
 
-In this guide, you'll learn about various methods of running your bot in a VPS, keeping it online 24/7, making it run automatically when your VPS boots and restart on crashes.
+## Server Rental
+
+> You need to rent a VPS in order to follow this guide.
+> This first section will explain how to do that.
+> If you already have a VPS to work on, skip [down](#starting-the-bot).
+
+We chose [Hostinger](https://hostinger.com/) as our host.
+There are several reasons for this:
+
+1. It is multilingual, so you will find it convenient and easy to use.
+2. It has a very rich documentation that thoroughly covers all the problem areas that a beginner may encounter.
+3. It offers powerful servers at fairly low prices.
+
+> You are free to choose the provider of your choice.
+> All providers provide the same services, so you won't have any problems with the technical part of this article.
+> You can take this part as an overview of how the server rental operation works.
+> If you are a beginner, you can use this guide to rent your first ever server!
+
+:::tip Analog of a server
+If you cannot or do not want to rent a server, you can replace it with a virtual machine.
+To do this, use an application such as [VirtualBox](https://virtualbox.org/).
+Create a virtual machine with the desired Linux distribution and use it as a real server.
+:::
```

Comment:
To be honest, I see no reason why this would be helpful to anyone. A VM running on a local machine cannot be used for hosting, so people must rent a server anyway. VMs are useful if you rent bare metal in a time before Docker was invented. I do not think our readers fall into that category.

I suggest to remove this tip entirely.

Reply:
And what if a person needs to gain experience in deploying to a remote server, but cannot rent it? Then a virtual machine will do the trick.

Reply:
Then we should mention that:
```suggestion
:::tip Analog of a server
If you cannot or do not want to rent a server but you still want to play around with running a bot on a VPS, you can follow this tutorial on a virtual machine instead.
To do this, use an application such as [VirtualBox](https://virtualbox.org/).
Create a virtual machine with the desired Linux distribution and use it as a real server.
:::
```